vars/buildAssembleUpload.groovy

```diff
@@ -12,29 +12,42 @@ void call(Map args = [:]) {
     manifestSHA = sha1(manifestLock)
     echo "Manifest SHA: ${manifestSHA}"
 
-    lib = library(identifier: 'jenkins@20211123', retriever: legacySCM(scm))
+    def lib = library(identifier: 'jenkins@20211123', retriever: legacySCM(scm))
     def inputManifest = lib.jenkins.InputManifest.new(readYaml(file: manifestLock))
     String shasRoot = inputManifest.getSHAsRoot("${JOB_NAME}", args.platform, args.architecture)
+    String manifestShaPath = "${shasRoot}/${manifestSHA}.yml"
 
-    buildManifest(
-        args + [
-            manifest: manifestLock
-        ]
-    )
+    Boolean shaExists = false
+    withAWS(role: 'opensearch-bundle', roleAccount: "${AWS_ACCOUNT_PUBLIC}", duration: 900, roleSessionName: 'jenkins-session') {
+        if (!args.dryRun && s3DoesObjectExist(bucket: "${ARTIFACT_BUCKET_NAME}", path: manifestShaPath)) {
+            shaExists = true
+        }
+    }
 
-    assembleManifest(
-        args + [
-            manifest: args.dryRun ? 'tests/data/opensearch-build-1.1.0.yml' : 'builds/opensearch/manifest.yml'
-        ]
-    )
+    if (shaExists) {
+        lib.jenkins.Messages.new(this).add("${STAGE_NAME}", "Skipped ${STAGE_NAME}, ${manifestShaPath} exists.")
```

Comment:
This will show up in the slack notification message, seems like if that is the case we would sent a different notification or not send one at all

Reply:
We collect messages in the end and I believe would fail if there's no message from this stage, so this way the status in Slack shows clearly that the build was skipped for this stage.